src/socket/ip_socket.cr

```diff
@@ -1,139 +1,25 @@
 class IPSocket < Socket
   # Returns the `IPAddress` for the local end of the IP socket.
   def local_address
-    sockaddr = uninitialized LibC::SockaddrIn6
+    sockaddr = Pointer(LibC::SockaddrIn6).malloc.as(LibC::Sockaddr*)
```

Comment:
No worries, we can take care of this later. It's just a tiny allocation and only used if you do request the local address :-)